Doctrine/Orm/Filter/AbstractFilter.php

```diff
@@ -69,7 +72,111 @@ protected function getClassMetadata(ResourceInterface $resource)
      */
     protected function isPropertyEnabled($property)
     {
-        return null === $this->properties || array_key_exists($property, $this->properties);
+        $enabled = false;
+
+        if (null === $this->properties) {
+            // to ensure sanity, nested properties must still be explicitly enabled
+            $enabled = !$this->isPropertyNested($property);
```

Comment:
Hmm re-reading the code it makes sense to me now. The case of `$this->properties` being null is indeed an exceptional case.